stdlib/type.mli

```diff
@@ -35,3 +35,85 @@ type (_, _) eq = Equal: ('a, 'a) eq
 
     At runtime, this function simply returns its second argument unchanged.
 *)
+
+(** {1:identifiers Type identifiers} *)
+
+(** Type identifiers.
+
+    A type identifier is a value that denotes a type. Given two type
+    identifiers, they can be tested for {{!Id.provably_equal}equality} to
+    prove they denote the same type. Note that:
+
+    - Unequal identifiers do not imply unequal types: a given type can be
+      denoted by more than one identifier.
+    - Type identifiers can be marshalled, but they get a new, distinct,
+      identity on unmarshalling, so the equalities are lost.
+
+    See an {{!Id.example}example} of use. *)
+module Id : sig
+
+  (** {1:ids Type identifiers} *)
+
+  type 'a t
```

Comment:
Seems like this could be `type !'a t` and that might help?

Reply:
Indeed, but see the resolved discussion above.